subsys/bluetooth/mesh/cdb.c

```diff
@@ -1024,26 +1024,32 @@ int bt_mesh_cdb_node_key_export(const struct bt_mesh_cdb_node *node, uint8_t out
 	return bt_mesh_key_export(out, &node->dev_key);
 }
 
-struct bt_mesh_cdb_app_key *bt_mesh_cdb_app_key_alloc(uint16_t net_idx,
-						      uint16_t app_idx)
+struct bt_mesh_cdb_app_key *bt_mesh_cdb_app_key_alloc(uint16_t net_idx, uint16_t app_idx)
 {
 	struct bt_mesh_cdb_app_key *key;
+	struct bt_mesh_cdb_app_key *vacant_key = NULL;
 	int i;
 
 	for (i = 0; i < ARRAY_SIZE(bt_mesh_cdb.app_keys); ++i) {
 		key = &bt_mesh_cdb.app_keys[i];
 
-		if (key->net_idx != BT_MESH_KEY_UNUSED) {
+		if (key->app_idx == app_idx) {
+			return NULL;
```

Comment:
Update function description?

Reply:
What exactly do you want me to change? Just add that it does not allow duplicate keys?

Looking at `bt_mesh_cdb_subnet_alloc` docs it dose not specify that duplicate netkey are not allowed.

Reply:
> Just add that it does not allow duplicate keys?

Yes.

> Looking at `bt_mesh_cdb_subnet_alloc` docs it dose not specify that duplicate netkey are not allowed.

Bad function description really.

Reply:
Ok, I'll fix that one as well then 👍

Reply:
We see this from stack developers perspective. But users rely on the documentation, not on the implementation. As a user I shouldn't even look at how a function is implemented. If it is not stated that NULL can be because the key is already added, I will be confused and start looking at for example how big is the app keys array in the CDB instead of checking if that key is already added. The documentation is also a contract. Now it doesn't say anything about duplicate key, which means we could also return the already allocated key. By stating this in the doc we fixate this behavior of the function. This helps users to rely on the function behavior from release to release (until of course we decide to change this API).